utils/httpUtils.ts

```diff
@@ -72,3 +71,112 @@ export const notFound = (message: string) => {
+  let cleanQuery = query;
+  // remove fed Prefix
+  const firstCurlyBracket = query.indexOf("{");
+  const secondCurlyBracket = query.indexOf("{", firstCurlyBracket + 1);
+  const fedIndex = query.indexOf("fed");
+  if (fedIndex !== -1 && fedIndex < secondCurlyBracket && fedIndex > firstCurlyBracket) {
+    const splitOnFirstFed = [query.substring(0, fedIndex + 4), query.substring(fedIndex + 4)];
+    const splitQueryOnFed = splitOnFirstFed[0].split("fed");
+    const firstLetterLowerCase = splitQueryOnFed[1][0].toLowerCase();
+    splitQueryOnFed[1] = splitQueryOnFed[1].slice(1);
+    splitQueryOnFed.splice(1, 0, firstLetterLowerCase);
+    cleanQuery = splitQueryOnFed.join("").concat(splitOnFirstFed[1]);
+  }
+
+  // remove input object from variables
+  const finishedQuery = cleanQuery
+    .replace("input: {", "")
+    .replace("}}", "}")
+    // apply any specific type switches that need to be made
+    .replace("$workflowRunId: String", "$workflowRunId: UUID")
+    .replace(/query_fedConsensusGenomes_items/g, "ConsensusGenome");
+
+  return finishedQuery;
+};
+
+export const fetchFromNextGen = async ({
+  args,
+  context,
+  serviceType,
+  fullResponse,
+  customQuery,
+}: {
+  args;
+  context;
+  serviceType: "workflows" | "entities";
+  fullResponse?: boolean;
+  customQuery?: string;
+}) => {
+  const enrichedToken = await getEnrichedToken(context);
+  const baseUrl = serviceType === "workflows" ? process.env.NEXTGEN_WORKFLOWS_URL : process.env.NEXTGEN_ENTITIES_URL;
+  const formattedQuery = customQuery ? customQuery : formatFedQueryForNextGen(context.params.query);
+  const response = await fetch(`${baseUrl}/graphql`, {
+    method: "POST",
+    headers: {
+      Cookie: context.request.headers.get("cookie"),
+      "Content-Type": "application/json",
+      "X-CSRF-Token": args?.input?.authenticityToken,
+      Authorization: `Bearer ${enrichedToken}`,
+    },
+    body: JSON.stringify({
+      query: formattedQuery,
+      variables: context.params.variables,
+    }),
+  });
+  if (fullResponse === true) {
+    return response;
+  } else {
+    return await response.json();
+  }
+};
+
+export const getFromRails = async ({
+  url,
+  args,
+  context,
+  fullResponse,
+}: {
+  url: string;
+  args: any;
+  context: any;
+  fullResponse?: boolean;
+}) => {
+  const baseURL = process.env.API_URL;
+  const urlPrefix = args.snapshotLinkId ? `/pub/${args.snapshotLinkId}` : "";
+
+  const response = await fetch(baseURL + urlPrefix + url, {
+    method: "GET",
+    headers: {
+      Cookie: context.request.headers.get("cookie"),
+      "Content-Type": "application/json",
+    },
+  });
+  if (fullResponse === true) {
+    return response;
+  } else {
+    return await response.json();
+  }
+};
+
+const checkForLogin = (responseUrl: string | null) => {
+  if (responseUrl?.includes("/auth0/refresh_token?mode=login")) {
+    throw new Error("You must be logged in to perform this action.");
```

Comment:
I think in the past, if you throw an error and it isn't caught, it will kill the mesh process this code was running in (although it automatically attempts to restart it).  Looking above, I believe this error should be caught and we reject the Promise, right?  Just wanted to verify so we don't kill the server processes 😅

Reply:
Mmm yeah, this function has been there for a while but now that the functions are modular maybe I need try catch statements in the fetchFromNextGen and the getFromRails incase they are called on their own